File: modules/agent/src/models/Comment.py
"""Comment model for polymorphic comments across entities."""

from datetime import datetime
from sqlalchemy import Column, Integer, String, Text, DateTime, ForeignKey
from sqlalchemy.orm import relationship
from models import Base


class Comment(Base):
    __tablename__ = "Comment"

    id = Column(Integer, primary_key=True)
    tenant_id = Column(Integer, ForeignKey("Tenant.id", ondelete="CASCADE"), nullable=False)
    commentable_type = Column(String(50), nullable=False)
    commentable_id = Column(Integer, nullable=False)
    content = Column(Text, nullable=False)
    created_by = Column(Integer, ForeignKey("User.id"), nullable=False)
    updated_by = Column(Integer, ForeignKey("User.id"), nullable=False)
    parent_comment_id = Column(Integer, ForeignKey("Comment.id", ondelete="CASCADE"), nullable=True)
    created_at = Column(DateTime(timezone=True), default=datetime.utcnow)
    updated_at = Column(DateTime(timezone=True), default=datetime.utcnow, onupdate=datetime.utcnow)

    # Relationships
    tenant = relationship("Tenant", back_populates="comments")
    creator = relationship("User", foreign_keys=[created_by])
    parent = relationship("Comment", remote_side=[id], backref="replies")

    def __repr__(self):
        return f"<Comment(id={self.id}, commentable_type={self.commentable_type}, commentable_id={self.commentable_id})>"

File: modules/agent/internal/models/Comment.go
package models

import "time"

type Comment struct {
	ID              int64     `gorm:"primaryKey;autoIncrement" json:"id"`
	TenantID        int64     `gorm:"not null;index" json:"tenant_id"`
	CommentableType string    `gorm:"size:50;not null" json:"commentable_type"`
	CommentableID   int64     `gorm:"not null" json:"commentable_id"`
	Content         string    `gorm:"type:text;not null" json:"content"`
	ParentCommentID *int64    `gorm:"index" json:"parent_comment_id"`
	CreatedBy       int64     `gorm:"not null" json:"created_by"`
	UpdatedBy       int64     `gorm:"not null" json:"updated_by"`
	CreatedAt       time.Time `gorm:"autoCreateTime" json:"created_at"`
	UpdatedAt       time.Time `gorm:"autoUpdateTime" json:"updated_at"`

	// Relationships
	Creator *User `gorm:"foreignKey:CreatedBy" json:"creator,omitempty"`
}

func (Comment) TableName() string {
	return "Comment"
}
